### PR TITLE
feat: trlst-216 Allow to remove non-confidential files without a confidential version.

### PR DESCRIPTION
**Pull request by Luisella21**

### CODE REVIEW

**Comment by Luisella21** [comment]
Normally a public user can only upload a confidential file, followed by the corresponding non-confidential version.
A user managed to upload an extra non-confidential file (possibly by uploading from two different windows). 
The extra file is stops him from completing the submission, but there is no provision to remove it.
Added an extra category when showing the uploaded file. The category lists the single non-confidential version, and ask the user to remove it. The interface allows to download the file before deleting it.
image

File: trade_remedies_public/cases/utils.py [hunk 1]
```diff
@@ -106,6 +106,8 @@ def structure_documents(documents):
     type_idx = deep_index_items_by(documents, "type/key")
     conf_idx = deep_index_items_by(type_idx.get("respondent") or [], "confidential")
     doc_idx = deep_index_items_by(documents, "id")
+    parent_idx = deep_index_items_by(conf_idx.get("false", []), "parent_id")
```
Comment (rossmiller) [inline trade_remedies_public/cases/utils.py:109]: Could you give this a more meaningful name?

**Review by rossmiller — DISMISSED** [review]
One very minor point!

File: trade_remedies_public/cases/utils.py (hunk 1 above)
Comment (davecharles) [reply trade_remedies_public/cases/utils.py:109]: So maybe:
```python
document_types = ...
confidential_docs = ...
all_docs_by_id = ...
doc_parents = ...
```

**Review by davecharles — CHANGES_REQUESTED** [review]
See comments

File: trade_remedies_public/cases/utils.py (hunk 1 above)
Comment (Luisella21) [reply trade_remedies_public/cases/utils.py]: Thank you Dave, very good suggestion!

**Review by davecharles — APPROVED** [review]
LGTM

**Pull request merged**